File: frontend/src/api/color.ts
import request from './request';
import { PagedResult } from './user';

// ── 类型 ──
export interface Color {
  id: string;
  code: string;
  nameZh: string;
  nameEn: string;
  hex: string;
  colorFamily: string;
  remark?: string;
  sortOrder: number;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateColorRequest {
  code: string;
  nameZh: string;
  nameEn: string;
  hex: string;
  colorFamily: string;
  remark?: string;
  sortOrder: number;
}

export interface UpdateColorRequest {
  nameZh?: string;
  nameEn?: string;
  hex?: string;
  colorFamily?: string;
  remark?: string;
  sortOrder?: number;
}

// ── 请求函数 ──
export function getColors(params: {
  page?: number;
  pageSize?: number;
  keyword?: string;
  colorFamily?: string;
  isActive?: boolean;
}) {
  return request.get<unknown, PagedResult<Color>>('/api/colors', { params });
}

export function getAllActiveColors() {
  return request.get<unknown, Color[]>('/api/colors/all');
}

export function getColor(id: string) {
  return request.get<unknown, Color>(`/api/colors/${id}`);
}

export function createColor(data: CreateColorRequest) {
  return request.post<unknown, Color>('/api/colors', data);
}

export function updateColor(id: string, data: UpdateColorRequest) {
  return request.put(`/api/colors/${id}`, data);
}

export function updateColorStatus(id: string, isActive: boolean) {
  return request.put(`/api/colors/${id}/status`, { isActive });
}

File: backend/src/OneCup.Domain/Entities/Color.cs
namespace OneCup.Domain.Entities;

/// <summary>
/// 颜色主数据字典。code 创建后不可改，作为面料/产品等业务模块的稳定引用标识符。
/// </summary>
public class Color : BaseEntity
{
    /// <summary>编码，如 RED001。创建后不可改</summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>中文名，如"大红"</summary>
    public string NameZh { get; set; } = string.Empty;

    /// <summary>英文名，如"Red"</summary>
    public string NameEn { get; set; } = string.Empty;

    /// <summary>颜色值 #RRGGBB</summary>
    public string Hex { get; set; } = string.Empty;

    /// <summary>颜色系（自由文本，如"红"）</summary>
    public string ColorFamily { get; set; } = string.Empty;

    /// <summary>备注</summary>
    public string? Remark { get; set; }

    /// <summary>排序号</summary>
    public int SortOrder { get; set; }

    /// <summary>启停状态（停用后引用方按需处理，不物理删除）</summary>
    public bool IsActive { get; set; } = true;
}
